File: src/main/java/com/dongfg/project/api/config/ApiProperty.java
package com.dongfg.project.api.config;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

/**
 * @author dongfg
 * @date 17-11-22
 */
@Data
@Configuration
@ConfigurationProperties(prefix = "api")
public class ApiProperty {
    /**
     * 云片短信发送
     */
    private String yunpianApikey;

    /**
     * chrome消息推送
     */
    private String privateVapid;

    /**
     * 两步验证KEY
     */
    private String otpSecret;
}

File: src/main/kotlin/com/dongfg/project/api/common/config/ApiProperty.kt
package com.dongfg.project.api.common.config

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.Configuration

/**
 * @author dongfg
 * @date 2018/3/16
 */
@Configuration
@ConfigurationProperties(prefix = "api")
class ApiProperty {
    lateinit var yunpianApikey: String
}